### DISCUSSION TITLE
Support comments at the end of keybind config line

### DISCUSSION THREAD

**Discussion by tekumara** [ghostty-org/ghostty#2699] [Feature Requests, Ideas] [answered]
eg:
```
keybind = alt+left=esc:b     # ⌥← backward word
keybind = alt+right=esc:f    # ⌥→ forward word
```

Currently comments will be output in key press, instead of ignored, which is surprising.

If this is the expected behaviour, it would be good to at least document it.

**Comment by mitchellh** [comment, marked as answer]
From the README:

```
# Comments start with a `#` and are only valid on their own line.
```

This is intentional and documented.

A brief why: if we get into trailing comments, we also have to get into quoted strings and ways to include `#` in a value. Right now, our parsing is extremely simple because we can simply read the full line as-is. It's not a _ton_ of complexity to parse a quoted line ignoring a trailing comment but its more complexity nonetheless. 

I'm open to adding this complexity eventually but want to see more support for it. For the time being, only a handful of people out of thousands of testers have raised this concern.

**Reply by tekumara** [reply]
Thank you for the explanation - that makes sense.

I missed that in the README. Instead I was looking in `ghostty +show-config --default --docs` for info about comments, and didn't see mention of this behaviour there.
